package_buildpack.go

```diff
@@ -51,31 +51,47 @@ func (c *Client) PackageBuildpack(ctx context.Context, opts PackageBuildpackOpti
 
 	packageBuilder.SetBuildpack(bp)
 
+	var bps []dist.Buildpack
 	for _, dep := range opts.Config.Dependencies {
 		if dep.URI != "" {
 			blob, err := c.downloader.Download(ctx, dep.URI)
 			if err != nil {
 				return errors.Wrapf(err, "downloading buildpack from %s", style.Symbol(dep.URI))
 			}
 
-			depBP, err := dist.BuildpackFromRootBlob(blob, archive.DefaultTarWriterFactory())
+			isOCILayout, err := buildpackage.IsOCILayoutBlob(blob)
 			if err != nil {
-				return errors.Wrapf(err, "creating buildpack from %s", style.Symbol(dep.URI))
+				return errors.Wrap(err, "inspecting buildpack blob")
 			}
 
-			packageBuilder.AddDependency(depBP)
+			if isOCILayout {
+				mainBP, depBPs, err := buildpackage.BuildpacksFromOCILayoutBlob(blob)
+				if err != nil {
+					return errors.Wrapf(err, "extracting buildpacks from %s", style.Symbol(dep.URI))
+				}
+
+				bps = append([]dist.Buildpack{mainBP}, depBPs...)
+			} else {
+				depBP, err := dist.BuildpackFromRootBlob(blob, archive.DefaultTarWriterFactory())
+				if err != nil {
+					return errors.Wrapf(err, "creating buildpack from %s", style.Symbol(dep.URI))
+				}
+				bps = []dist.Buildpack{depBP}
```

Comment:
All these `bps` assignments should be taking into consideration any prior added buildpacks to the array.

Such as:
```suggestion
				bps = append(bps, depBP)
```

For example, the way I'm reading this, if a directory buildpack were to be last in the list of dependencies it would drop all prior `bps` entries.

Reply:
Thanks for pointing that out! I'm adding a test to verify that that doesn't happen in the future